hayeon/baekjoon/수학/11050-이항-계수-1.js

```diff
@@ -0,0 +1,39 @@
+const readline = require("readline");
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+
+const input = [];
+
+rl.on("line", function (line) {
+  input.push(line);
+}).on("close", function () {
+  // 풀이
+  const [naturalNum, integer] = input[0].split(" ").map(Number);
+  let numerator = 1;
+  let denominator = 1;
+  let subtraction = 1;
+  let result = 0;
+
+  for (let i = 1; i <= naturalNum; i++) {
+    numerator *= i;
+  }
+  for (let j = 1; j <= integer; j++) {
+    denominator *= j;
+  }
+  for (let k = 1; k <= naturalNum - integer; k++) {
+    subtraction *= k;
+  }
```

Comment:
```suggestion
  let numerator = 1;
  let denominator = 1;
  let subtraction = 1;
  let result = 0;

  for (let i = 1; i <= naturalNum; i++) {
    numerator *= i;
  }
  for (let j = 1; j <= integer; j++) {
    denominator *= j;
  }
  for (let k = 1; k <= naturalNum - integer; k++) {
    subtraction *= k;
  }
```
- 풀이 잘봤습니다 🙇‍♂️
- 비슷한 로직이 반복적으로 하드코딩되있는게 가독성이 안좋은 것 같아요  해당 for문 반복되는걸 함수로 빼서 개선할 수 없을까요?? 그리고 for문 내에 사용하는 i, j, k는 중첩 for문시에 사용하는거고 같은 레벨이면 그냥 i사용하셔도 무방합니다.
- 또한 단순 반복문이 아닌 재귀함수를 이용해서 팩토리얼을 구하는 로직도 존재합니다 
```js
// ex)
for(let i = 0; i < length; i++){
  for(let j = 0; j < length; j++){}
}

for(let i = 0; i < length; i++){}
```

Reply:
오 팩토리얼 로직으로 구하는 법 코드 참고했는데 훨씬 간단하네요!!!!!!!😮